### PR TITLE
More Unhandled Exception Information

### PR DESCRIPTION
**Pull request by dagrooms52**
- Add stack trace to unhandled exception logs
- Add site's current route to unhandled exception logs

### CODE REVIEW

File: AngularApp/projects/diagnostic-data/src/lib/services/unhandled-exception-handler.service.ts [hunk 1]
```diff
@@ -7,14 +8,28 @@ import { KustoTelemetryService } from './telemetry/kusto-telemetry.service';
...
+
+            const props = {
+                'route': this.router.url
+            }
+
+            if (error.stack != undefined) {
+                props['stack'] = error.stack;
+            }
+
+            this.logService.logException(error, "unhandled", props, null, SeverityLevel.Critical);
```
Comment (ShekharGupta1988) [inline AngularApp/projects/diagnostic-data/src/lib/services/unhandled-exception-handler.service.ts]: Can you put an example how the event looks like now?

File: AngularApp/projects/diagnostic-data/src/lib/services/unhandled-exception-handler.service.ts [hunk 2]
```diff
@@ -7,14 +8,28 @@ import { KustoTelemetryService } from './telemetry/kusto-telemetry.service';
...
-      // Squash logging error
+    constructor(private logService: KustoTelemetryService, private injector: Injector) { }
+
+    handleError(error: Error) {
+        try {
+            if (this.router == undefined) {
+                this.router = this.injector.get(Router);
+            }
+
+            const props = {
```
Comment (cindylovescoding) [inline AngularApp/projects/diagnostic-data/src/lib/services/unhandled-exception-handler.service.ts]: Should we define this into const if you are going to add more key value pair below?

Comment (dagrooms52) [reply AngularApp/projects/diagnostic-data/src/lib/services/unhandled-exception-handler.service.ts]: Yes, I'm only adding keys not repointing `props` to a different object

File: AngularApp/projects/diagnostic-data/src/lib/services/unhandled-exception-handler.service.ts (hunk 1 above)
Comment (dagrooms52) [reply AngularApp/projects/diagnostic-data/src/lib/services/unhandled-exception-handler.service.ts]: Sure, here's the exception object. You can see I added `route` and `stack` to the logs. I cut it off in the screenshot but the route goes to "{resourceUri}/categories/WindowsAvailabilityAndPerformance", and the stack is verbose although it only details internal Angular code in this case. I've tested that it does show stack traces for our code as well.

exception_logging

**Pull request merged**